ksqldb-rest-app/src/test/java/io/confluent/ksql/api/ProxyProtocolTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.startsWith;
+
+import io.confluent.ksql.rest.server.KsqlRestConfig;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.net.Socket;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.Test;
+
+public class ProxyProtocolTest extends ApiTest {
+
+  @Override
+  protected KsqlRestConfig createServerConfig() {
+    final Map<String, Object> config = new HashMap<>();
+    config.put(KsqlRestConfig.LISTENERS_CONFIG, "http://localhost:8080,http://localhost:8081");
+    config.put(KsqlRestConfig.PROXY_PROTOCOL_LISTENERS_CONFIG, "http://localhost:8081");
+    config.put(KsqlRestConfig.VERTICLE_INSTANCES, 4);
+    return new KsqlRestConfig(config);
+  }
+
+  @Test
+  public void shouldHandleRequestsOnProxyProtocolListener() throws Exception {
```

Comment:
Thanks for adding this integration test! This is awesome.

What's the expected behavior if a request without the PROXY header is send to the PROXY protocol listener, or vice versa? Sounds like it fails but hopefully in a way that's clear to the user what's going on. (We don't necessarily need to add tests for this if you don't think it makes sense; as long as you've thought through the implications I'm fine with that.)

Reply:
I added additional tests for these two cases. PROXY protocol drops the connection, HTTP returns 400 Bad Request